server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsServiceImpl.java

```diff
@@ -481,17 +478,11 @@ public Long getDelay() {
 
         private void cleanupOldDiagnosticFiles(DataStore store) {
             String mountPoint = null;
-            try {
-                mountPoint = serviceImpl.mountManager.getMountPoint(store.getUri(), null);
-                if (StringUtils.isNotBlank(mountPoint)) {
-                    File directory = new File(mountPoint + File.separator + DIAGNOSTICS_DIRECTORY);
-                    if (directory.isDirectory()) {
-                        deleteOldDiagnosticsFiles(directory, store.getName());
-                    }
-                }
-            } finally {
-                if (StringUtils.isNotBlank(mountPoint)) {
-                    umountSecondaryStorage(mountPoint);
+            mountPoint = serviceImpl.mountManager.getMountPoint(store.getUri(), null);
+            if (StringUtils.isNotBlank(mountPoint)) {
+                File directory = new File(mountPoint + File.separator + DIAGNOSTICS_DIRECTORY);
+                if (directory.isDirectory()) {
+                    deleteOldDiagnosticsFiles(directory, store.getName());
```

Comment:
Does the secondary storage stay mounted on hypervisors other than VMware, after cleanup of diagnostics files? If so, it is better to fix the permission issue (on VMware), and unmount the secondary storage from mgmt server when not needed.

Reply:
@Pearl1594 any update on this ^^^ , possible to fix this permission issue for VMware only?

Reply:
Sorry @sureshanaparti I seemed to have missed your earlier comment. Currently what's happening is that due to the permission issue, we see the secondary stores remain mounted on all the hypervisors. We could fix the permission issue using this solution - https://github.com/shapeblue/cloudstack/commit/a9b460869b8f244f4b3db3cc3973f3419dffed83. However, I am of the opinion that we probably don't need to mount and un-mount every time the garbage collection thread runs. As fixing the permission issue wouldn't solve the issue entirely, we'd also need to cleanup the mount path (at /var/cloudstack/mnt) and remove that mount path from the static list of storageMounts maintained, so that in successive runs are successful.

Reply:
OK @Pearl1594 , Got it. I think, keeping secondary store mounted always on all the hypervisors is also not good. Only SSVM should have the access to it always.

Reply:
I would agree if this was on hosts @sureshanaparti , but it is on the MS and
1. I would hate using sudo once more in the system
2. I would prefer to keep as much uniformity over hypervisor types as possible, so rather not have a vmware specific solution.
what do you think?

Reply:
> I would agree if this was on hosts @sureshanaparti , but it is on the MS and
> 
> 1. I would hate using sudo once more in the system
> 2. I would prefer to keep as much uniformity over hypervisor types as possible, so rather not have a vmware specific solution.
>    what do you think?

agree @DaanHoogland